recipe/build.sh

```diff
@@ -1,19 +1,20 @@
 #!/bin/bash
 
-if [[ "${cuda_compiler_version}" =~ 12.* ]]; then
-  export CUDA_HOME=${BUILD_PREFIX}
-elif [[ "${cuda_compiler_version}" != "None" ]]; then
-  export CUDA_HOME=${CUDA_PATH}
+if [[ -z "${CUDA_HOME+x}" ]]; then
+  # `$CUDA_HOME` was set for CUDA 11.x and earlier.
+  # Must be using CUDA 12.0+. So set to `$BUILD_PREFIX`.
+  # This is needed to find `nvcc` and `cudart`.
+  export CUDA_HOME="${BUILD_PREFIX}"
 fi
 
-if [[ $CONDA_BUILD_CROSS_COMPILATION == "1" ]]; then
-    if [[ $target_platform == linux-aarch64 ]]; then
-        export CUDA_LIB=${CUDA_HOME}/targets/sbsa-linux/lib/
-    elif [[ $target_platform == linux-ppc64le ]]; then
-        export CUDA_LIB=${CUDA_HOME}/targets/ppc64le-linux/lib/
+if [[ "${CONDA_BUILD_CROSS_COMPILATION}" == "1" ]]; then
+    if [[ "${target_platform}" == "linux-aarch64" ]]; then
+        export CUDA_LIB="${CUDA_HOME}/targets/sbsa-linux/lib/"
+    elif [[ "${target_platform}" == "linux-ppc64le" ]]; then
+        export CUDA_LIB="${CUDA_HOME}/targets/ppc64le-linux/lib/"
     else
         echo "not supported"
-        exit -1
+        exit 1
```

Comment:
Shellcheck didn't like `exit -1` as it was invalid. It said we should use a value in `0-255`. So picked `1`